File: app/src/main/java/me/liuyun/bjutlgn/api/BjutService.java
package me.liuyun.bjutlgn.api;

import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface BjutService {
    @FormUrlEncoded
    @POST("/")
    Observable<String> login(@Field("DDDDD") String user, @Field("upass") String pass, @Field("R6") String r6, @Field("6MKKey") String out);

    @FormUrlEncoded
    @POST("/")
    Observable<String> loginLocal(@Field("DDDDD") String user, @Field("upass") String pass, @Field("R6") String r6, @Field("0MKKey") String in);

    @GET("/F.htm")
    Observable<String> logout();

    @GET("/1.htm")
    Observable<String> stats();
}

File: app/src/main/kotlin/me/liuyun/bjutlgn/api/BjutService.kt
package me.liuyun.bjutlgn.api

import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST

interface BjutService {
    @FormUrlEncoded
    @POST("/")
    fun login(@Field("DDDDD") user: String, @Field("upass") pass: String, @Field("R6") r6: String, @Field("6MKKey") out: String): Observable<String>

    @FormUrlEncoded
    @POST("/")
    fun loginLocal(@Field("DDDDD") user: String, @Field("upass") pass: String, @Field("R6") r6: String, @Field("0MKKey") `in`: String): Observable<String>

    @GET("/F.htm")
    fun logout(): Observable<String>

    @GET("/1.htm")
    fun stats(): Observable<String>
}